src/Features/Core/Portable/CodeFixes/AddImport/AbstractAddImportCodeFixProvider.SymbolReference.cs

```diff
@@ -1,99 +1,104 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
-using Microsoft.CodeAnalysis.Shared.Extensions;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Editing;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeFixes.AddImport
 {
     internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSyntax>
     {
-        private abstract class SymbolReference : IComparable<SymbolReference>, IEquatable<SymbolReference>
+        private abstract class SymbolReference : Reference
         {
-            public readonly SearchResult<INamespaceOrTypeSymbol> SearchResult;
+            public readonly SymbolResult<INamespaceOrTypeSymbol> SymbolResult;
 
-            protected SymbolReference(SearchResult<INamespaceOrTypeSymbol> searchResult)
+            public SymbolReference(AbstractAddImportCodeFixProvider<TSimpleNameSyntax> provider, SymbolResult<INamespaceOrTypeSymbol> symbolResult)
+                : base(provider, new SearchResult(symbolResult))
             {
-                this.SearchResult = searchResult;
+                this.SymbolResult = symbolResult;
             }
 
-            public int CompareTo(SymbolReference other)
-            {
-                // If references have different weights, order by the ones with lower weight (i.e.
-                // they are better matches).
-                if (this.SearchResult.Weight < other.SearchResult.Weight)
-                {
-                    return -1;
-                }
+            protected abstract Solution UpdateSolution(Document newDocument);
+            protected abstract Glyph? GetGlyph(Document document);
 
-                if (this.SearchResult.Weight > other.SearchResult.Weight)
+            public override int CompareTo(Reference other)
```

Comment:
Then is `IComparer` appropraite here?  This type's implementation of the interfaces do not match up with the guidelines. Our current use may be appropriate but leaving these markers are bad trail markers for future developers.